### PR TITLE
Show warning when auth fails

### PR DESCRIPTION
**Pull request by mymindstorm**

### CODE REVIEW

File: src/models/backup.ts [hunk 1]
```diff
@@ -157,6 +166,11 @@ class Drive {
 
   private async getFolder() {
     const token = await this.getToken();
+    if (!token) {
+      return new Promise((resolve: (value: false) => void) => {
```
Comment (Sneezry) [inline src/models/backup.ts]: do you mean `value: boolean` ?

File: src/models/backup.ts [hunk 2]
```diff
@@ -250,6 +264,11 @@ class Drive {
     const backup = JSON.stringify(exportData, null, 2);
 
     const token = await this.getToken();
+    if (!token) {
+      return new Promise((resolve: (value: false) => void) => {
```
Comment (Sneezry) [inline src/models/backup.ts]: same here

File: src/models/backup.ts [hunk 3]
```diff
@@ -132,11 +136,16 @@ class Drive {
...
                   localStorage.removeItem('driveRefreshToken');
+                  localStorage.driveRevoked = true;
                   resolve(false);
                 }
                 try {
                   const res = JSON.parse(xhr.responseText);
                   if (res.error) {
+                    if (res.error === 'invalid_grant') {
+                      localStorage.removeItem('driveRefreshToken');
+                      localStorage.driveRevoked = true;
```
Comment (Sneezry) [inline src/models/backup.ts:147]: do we need to clear token cache here?

Comment (mymindstorm) [reply src/models/backup.ts:147]: No, that only applies to getAuthToken (chrome only). Google doesn't give a 401 when the token gets revoked, and it wasn't actually deleteing them properly before.

File: src/models/backup.ts (hunk 1 above)
Comment (mymindstorm) [reply src/models/backup.ts]: Since it is only supposed to return false is it ok to just leave it like this or is it better to use Boolean?

Comment (Sneezry) [reply src/models/backup.ts]: `false` is not a type, `boolean` should be used here.

**Review by Sneezry — APPROVED** [review]

**Comment by bunchuairak** [comment]
eobot.com

ในวันที่ อ. 16 ต.ค. 2018 เวลา 09:38 Zhe Li <notifications@github.com>
เขียนว่า:

**Pull request merged**